src/algoliasearch.helper.js

```diff
@@ -431,10 +443,118 @@ AlgoliaSearchHelper.prototype.setState = function(newState) {
+  if (filters === undefined) return this.state;
+
+  var partialState = {};
+  var attributeFilters = filter(filters, function(f) { return f.indexOf('attribute:') !== -1; });
+  var attributes = map(attributeFilters, function(aF) { return aF.split(':')[1]; });
+  if (attributes.indexOf('*') === -1) {
+    forEach(attributes, function(attr) {
+      if (this.state.isConjunctiveFacet(attr) && this.state.isFacetRefined(attr)) {
+        if (!partialState.facetsRefinements) partialState.facetsRefinements = {};
+        partialState.facetsRefinements[attr] = this.state.facetsRefinements[attr];
+      }
+
+      if (this.state.isDisjunctiveFacet(attr) && this.state.isDisjunctiveFacetRefined(attr)) {
+        if (!partialState.disjunctiveFacetsRefinements) partialState.disjunctiveFacetsRefinements = {};
+        partialState.disjunctiveFacetsRefinements[attr] = this.state.disjunctiveFacetsRefinements[attr];
+      }
+
+      var numericRefinements = this.state.getNumericRefinements(attr);
+      if (!isEmpty(numericRefinements)) {
+        if (!partialState.numericRefinements) partialState.numericRefinements = {};
+        partialState.numericRefinements[attr] = this.state.numericRefinements[attr];
+      }
+    }, this);
+  } else {
+    if (!isEmpty(this.state.numericRefinements)) {
+      partialState.numericRefinements = this.state.numericRefinements;
+    }
+    if (!isEmpty(this.state.facetsRefinements)) partialState.facetsRefinements = this.state.facetsRefinements;
+    if (!isEmpty(this.state.disjunctiveFacetsRefinements)) {
+      partialState.disjunctiveFacetsRefinements = this.state.disjunctiveFacetsRefinements;
+    }
+    if (!isEmpty(this.state.hierarchicalFacetsRefinements)) {
+      partialState.hierarchicalFacetsRefinements = this.state.hierarchicalFacetsRefinements;
+    }
+  }
+
+  if (filters.indexOf('index') !== -1) {
+    partialState.index = this.index;
+  }
+
+  var searchParameters = filter(
+    filters,
+    function(f) { return f !== 'index' && f.indexOf('attribute:') === -1; });
+  forEach(searchParameters, function(parameterKey) {
+    partialState[parameterKey] = this.state[parameterKey];
+  }, this);
+
+  return partialState;
+};
+
+/**
+ * Get part of the
  */
-AlgoliaSearchHelper.prototype.getState = function() {
-  return this.state;
+AlgoliaSearchHelper.prototype.getStateAsQueryString = function getStateAsQueryString(filters, options) {
+  var moreAttributes = options && options.moreAttributes;
+  var prefixForParameters = options && options.prefix;
+
+  var filtersOrDefault = filters ? filters : ['query', 'attribute:*'];
+  var partialState = this.getState(filtersOrDefault);
+  var partialStateWithPrefix = prefixForParameters ?
+    mapKeys(
+      partialState,
+      function(v, k) { return prefixForParameters + k; }) :
+    partialState;
+
+  if (moreAttributes) merge(partialStateWithPrefix, moreAttributes);
+
+  return qs.stringify(partialStateWithPrefix);
```

Comment:
As array query string is not standardized, I dunno what is the support for these options.